### PR TITLE
Add memory error debugging description

### PR DESCRIPTION
**Pull request by negiyas**
Add the "Finding memory errors" section in docs/DebuggingNumericalError.md about how to use the '"Electric Fence library" for debugging.

### CODE REVIEW

File: docs/DebuggingNumericalError.md
Comment (negiyas) [reply docs/DebuggingNumericalError.md]: I updated the description. Thank for your comments.

Comment (tungld) [reply docs/DebuggingNumericalError.md]: It looks to me it is too detailed to list packages for different environments. There are many environments and we cannot list them all. For example, in the above list, packages are for fedora but not ubuntu, suse, etc. for linux, mac, but not windows.

Instead, we can suggest a list of tools that are useful to debug memory issues in onnx-mlir, e.g. valgrind, mtrace, efence. Which one is suggested to use in which situation, and some tips to use them in onnx-mlir, etc. Users should find and install the tools depending on their environments.

Comment (etiotto) [reply docs/DebuggingNumericalError.md]: > It looks to me it is too detailed to list packages for different environments. There are many environments and we cannot list them all. For example, in the above list, packages are for fedora but not ubuntu, suse, etc. for linux, mac, but not windows.

OK that is also a sensible solution. We should refrain from just listing a tool for s390x given that environment is just one of the possible environment for users of `onnx-mlir`.  List tools in a more generic way is fine by me.

Comment (negiyas) [reply docs/DebuggingNumericalError.md]: @etiotto @tungld  Thank you for your comments. I updated the description.
Your further comments and suggestions are very welcome.

File: docs/DebuggingNumericalError.md
Comment (etiotto) [inline docs/DebuggingNumericalError.md:118]: "malloc() debuggers" --> memory debuggers ?

**Review by etiotto — APPROVED** [review]
Thanks for the updated description @negiyas.

I think the information presented here is now in good shape, I left some suggestions on the grammar that  you can address before merging. 

LGTM

File: docs/DebuggingNumericalError.md
Comment (negiyas) [reply docs/DebuggingNumericalError.md:106]: Thank you. Changed to "that an onnx-mlir compiled"

File: docs/DebuggingNumericalError.md
Comment (negiyas) [reply docs/DebuggingNumericalError.md:107]: Thank you!

File: docs/DebuggingNumericalError.md
Comment (negiyas) [reply docs/DebuggingNumericalError.md:109]: done. Thank you!

File: docs/DebuggingNumericalError.md
Comment (negiyas) [reply docs/DebuggingNumericalError.md:110]: Done. Thank you!

File: docs/DebuggingNumericalError.md
Comment (negiyas) [reply docs/DebuggingNumericalError.md:112]: Done. Thank you.
I also replace the word "bug" to the word "problem."

File: docs/DebuggingNumericalError.md
Comment (negiyas) [reply docs/DebuggingNumericalError.md:113]: Done. Thank you.

File: docs/DebuggingNumericalError.md
Comment (negiyas) [reply docs/DebuggingNumericalError.md:116]: Done. Thank you.

File: docs/DebuggingNumericalError.md
Comment (negiyas) [reply docs/DebuggingNumericalError.md:118]: Done. Thank you.

File: docs/DebuggingNumericalError.md
Comment (negiyas) [reply docs/DebuggingNumericalError.md:123]: Changed to "your system" to on yours"

File: docs/DebuggingNumericalError.md
Comment (tungld) [inline docs/DebuggingNumericalError.md:108]: Please use link syntax in markdown, e.g. `[framework](https://valgrind.org/)`. The same for the other urls.

**Review by tungld — APPROVED** [review]
LGTM with pending link syntax.

File: docs/DebuggingNumericalError.md
Comment (negiyas) [reply docs/DebuggingNumericalError.md]: done. Thank you!

File: docs/DebuggingNumericalError.md
Comment (tungld) [inline docs/DebuggingNumericalError.md:108]: "framework" -> "valgrind framework"?

File: docs/DebuggingNumericalError.md
Comment (negiyas) [reply docs/DebuggingNumericalError.md]: done. Thank you.

File: docs/DebuggingNumericalError.md
Comment (negiyas) [reply docs/DebuggingNumericalError.md:108]: done. Thank you.

**Pull request merged**
